src/main/java/racingcar/domain/Car.java

```diff
@@ -0,0 +1,70 @@
+package racingcar.domain;
+
+import racingcar.utils.Constants;
+
+import java.util.Objects;
+
+public class Car implements Comparable<Car> {
+    private static final int INITIAL_POSITION = 0;
+    private static final int MINIMUM_NUMBER_TO_MOVE = 4;
+
+    private final Name name;
+    private final Position position;
+
+    Car(String name, int position) {
+        this.name = new Name(name);
+        this.position = new Position(position);
+    }
+
+    public Car(String name) {
+        this(name, INITIAL_POSITION);
+    }
+
+    public void moveOrStop(int number) {
+        validateNumber(number);
+
+        if (number >= MINIMUM_NUMBER_TO_MOVE) {
+            position.increase();
+        }
+    }
+
+    private void validateNumber(int number) {
+        if (number < Constants.MINIMUM_NUMBER_RANGE || number > Constants.MAXIMUM_NUMBER_RANGE) {
+            throw new IllegalArgumentException("잘못된 입력 값입니다.");
+        }
+    }
+
+    public boolean isSamePosition(Car car) {
+        return car.isMatchPosition(position);
+    }
+
+    private boolean isMatchPosition(Position position) {
+        return this.position.equals(position);
+    }
```

Comment:
```suggestion
    public boolean isSamePosition(Car car) {
        return position.equals(car.position);
    }
```
이렇게도 작성할 수 있을 거 같아요! 아니면 메서드를 분리하신 이유가 따로 있으실까요?? 😄

Reply:
`car.position`처럼 객체 필드에 직접 접근하지 않기 위함이였습니다!!
물론 작성해주신 코드도 깔끔하고 너무나 좋은 거 같아요!! 👍 감사합니다!!